packages/mobx-react-lite/src/utils/UniversalFinalizationRegistry.ts

```diff
@@ -0,0 +1,62 @@
+export declare class FinalizationRegistryType<T> {
```

Comment:
Would it be possible to do the FinalizationRegistery refactoring (makes lot of sense) in a separate PR? Then we could first rollout that as a patch change as, as we don't expect changes, and then separately roll out the change for React 18, to reduce the size of the change we have to debug if something goes awry? But if the changes are too intertwined I can live with this as well.

Reply:
PR is here https://github.com/mobxjs/mobx/pull/3598. However I've also changed `useObserver` impl a bit.